src/components/Register.js

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react"
+import { TextField, Button } from "@mui/material"
+import { Link } from "react-router-dom"
+import '../styles/Register.css'
+
+const Register = () => {
+    
+    const [createdUser, setCreatedUser] = useState(false)
+    const [username, setUserName] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+
+
+    const handleRegister = (event) => {
+        event.preventDefault()
+        if (username && email && password) {
+            const userObject = {
+                username,
+                email,
+                password
+            }
+            fetchRegister(userObject)
+        }
+    }
+
+    const fetchRegister = async (user) => {
+        const url = `https://recipe-blog-l7ey.onrender.com/recipe-blog/register`
+
+        const options = {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'username': user.username,
+                'email': user.email,
+                'password': user.password
+            })
+        }
+
+        try {
```

Comment:
Please make sure to check whether the API call was successful or not. If it was successful, set token to the local storage. However, if the API call was unsuccessful, set the error message to the error state. In the case of an error during the API call, set the error message to "Request failed". Here's the code for your reference:
```
try {
            const response = await fetch(url, options)

            const data = await response.json()
            
            if(response.ok) {
                localStorage.setItem('token', data.token)
                localStorage.setItem('user', data.user.id)
                setCreatedUser(true)
            } else {
                setErrorMessage(data.message)
            }
        } catch (error) {
            setErrorMessage("Request fail")
        }
```